rangeserver/src/range_manager.rs

```diff
@@ -418,13 +418,19 @@ where
                 if let Some(val) = value {
                     get_result.val = Some(val);
                 } else {
-                    let val = self
+                    // check prefetch buffer
+                    let value = self.prefetching_buffer.get_from_buffer(key.clone()).await;
+                    if let Some(val) = value {
+                        get_result.val = Some(val);
+                    } else {
+                        let val = self
```

Comment:
we always try to lookup the key if the buffer returns None, but might have information that the key does not exist in the database. We should distinguish the two cases: Buffer knows the key doesn't exist vs. Buffer does not have the key at all.